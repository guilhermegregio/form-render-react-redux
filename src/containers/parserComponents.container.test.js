import React from 'react';
import ReactDOM from 'react-dom';

import ParserComponentsContainer from './parserComponents.container';

const textField = {
    name: 'txtFullname',
    id: 'txtFullname',
    value: '',
    type: 'text',
    className: 'input'
};

const textAddressField = {
    name: 'txtAddress',
    id: 'txtAddress',
    value: '',
    type: 'text:address',
    className: 'input'
};

const buttonField = {
    type: 'button',
    name: 'btnSave',
    id: 'btnSave',
    value: 'Gravar',
    className: 'input'
};

const imageField = {
    type: 'image',
    name: 'imgAvatar',
    id: 'imgAvatar',
    className: 'image'
}

const uploadField = {
    type: 'upload',
    name: 'uplImage',
    id: 'uplImage',
    className: 'upload-button'
}

const mapField = {
    type: 'map',
    name: 'mapRenderer',
    id: 'mapRenderer',
    className: 'map-renderer'
};

it('default render parser component container', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParserComponentsContainer fields={[textField, textAddressField, mapField, uploadField, imageField, buttonField]} />, div);

    expect(div).toMatchSnapshot();
});
