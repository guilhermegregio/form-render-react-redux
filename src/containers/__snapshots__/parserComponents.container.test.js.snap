// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render parser component container 1`] = `
<div>
  <div
    data-reactroot=""
  >
    <div
      class="text-component group input"
    >
      <input
        type="text"
        id="txtFullname"
        name="txtFullname"
        placeholder=""
        required=""
        value=""
      />
      <span
        class="highlight"
      />
      <span
        class="bar"
      />
      <label>
        <!-- react-text: 7 -->
        txtFullname
        <!-- /react-text -->
        <!-- react-text: 8 -->
        :
        <!-- /react-text -->
      </label>
    </div>
    <div
      class="map-component group map-renderer"
    />
    <div
      class="upload-component group upload-button"
    >
      <input
        type="file"
        id="uplImage"
        name="uplImage"
      />
    </div>
    <div
      class="image-component group image"
    >
      <img
        name="imgAvatar"
        alt="imgAvatar"
      />
    </div>
    <div
      class="button-component group input"
    >
      <button
        id="btnSave"
        type="button"
        class="button primary"
        value="Gravar"
      >
        <span
          class="load fa fa-spinner"
        />
        <span
          class="value"
        >
          Gravar
        </span>
      </button>
    </div>
  </div>
</div>
`;
